File: common/database/src/main/java/com/lany192/box/database/AppDatabase.java
package com.lany192.box.database;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import com.lany192.box.database.entity.BrowseHistory;
import com.lany192.box.database.entity.SearchHistory;
import com.lany192.box.database.dao.BrowseHistoryDao;
import com.lany192.box.database.dao.SearchHistoryDao;


@Database(entities = {SearchHistory.class, BrowseHistory.class}, exportSchema = false, version = 1)
@TypeConverters({Converters.class})
public abstract class AppDatabase extends RoomDatabase {
    /**
     * 搜索记录
     */
    public abstract SearchHistoryDao searchHistoryDao();

    /**
     * 浏览记录
     */
    public abstract BrowseHistoryDao browseHistoryDao();
}

File: common/database/src/main/kotlin/com/lany192/box/database/AppDatabase.kt
package com.lany192.box.database

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.lany192.box.database.dao.BrowseHistoryDao
import com.lany192.box.database.dao.SearchHistoryDao
import com.lany192.box.database.entity.BrowseHistory
import com.lany192.box.database.entity.SearchHistory

@Database(
    entities = [SearchHistory::class, BrowseHistory::class],
    exportSchema = false,
    version = 1
)
@TypeConverters(
    Converters::class
)
abstract class AppDatabase : RoomDatabase() {
    /**
     * 搜索记录
     */
    abstract fun searchHistoryDao(): SearchHistoryDao

    /**
     * 浏览记录
     */
    abstract fun browseHistoryDao(): BrowseHistoryDao
}
